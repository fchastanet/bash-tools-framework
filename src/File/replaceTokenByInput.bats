#!/usr/bin/env bash

# shellcheck source=src/batsHeaders.sh
source "$(cd "${BATS_TEST_DIRNAME}/.." && pwd)/batsHeaders.sh"

# shellcheck source=/src/File/replaceTokenByInput.sh
source "${srcDir}/File/replaceTokenByInput.sh"
# shellcheck source=/src/Framework/createTempFile.sh
source "${srcDir}/Framework/createTempFile.sh"
# shellcheck source=/src/Filters/removeAnsiCodes.sh
source "${srcDir}/Filters/removeAnsiCodes.sh"

function File::replaceTokenByInput::OneTokenWithoutAnsiCodes { #@test
  cp "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput.txt" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"
  cat "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput.toInject.txt" | File::replaceTokenByInput \
    "@token@" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"

  diff "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput.expected.txt" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"
}

function File::replaceTokenByInput::OneTokenWithAnsiCodes { #@test
  cp "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput.txt" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"
  # cspell:disable
  (
    echo -e "\e[31mreplace\e[32mToken\e[1;30mByInput injected\e[7;49;33m"
  ) >"${BATS_TEST_TMPDIR}/replaceTokenByInput.toInject.txt"
  # cspell:enable
  cat "${BATS_TEST_TMPDIR}/replaceTokenByInput.toInject.txt" | File::replaceTokenByInput \
    "@token@" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"

  diff "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput.expected.txt" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"
}

function File::replaceTokenByInput::TwoTokens { #@test
  cp "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput2Tokens.txt" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput2Tokens.txt"
  cat "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput2Tokens.toInject.txt" | File::replaceTokenByInput \
    "@token@" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput2Tokens.txt"

  diff "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput2Tokens.expected.txt" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput2Tokens.txt" >&3
}

function File::replaceTokenByInput::NoMatchingPattern { #@test
  cp "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput.txt" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"
  cat "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput.toInject.txt" | File::replaceTokenByInput \
    "@tokenNotFound@" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"

  diff "${BATS_TEST_DIRNAME}/testsData/replaceTokenByInput.txt" \
    "${BATS_TEST_TMPDIR}/replaceTokenByInput.txt"
}
