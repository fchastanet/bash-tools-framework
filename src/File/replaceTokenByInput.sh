#!/usr/bin/env bash

File::replaceTokenByInput() {
  local token="$1"
  local targetFile="$2"

  (
    local tokenFile
    tokenFile="$(Framework::createTempFile "replaceTokenByInput")"

    cat - | Filters::escapeColorCodes >"${tokenFile}"

    sed -E -i \
      -e "/${token}/r ${tokenFile}" \
      -e "/${token}/d" \
      "${targetFile}"
  )
}
