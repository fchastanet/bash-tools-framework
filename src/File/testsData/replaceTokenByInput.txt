stuff before
@token@
stuff after
