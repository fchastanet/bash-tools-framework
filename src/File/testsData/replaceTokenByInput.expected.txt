stuff before
replaceTokenByInput injected
stuff after
